eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -166,18 +166,14 @@ conwayDelegTransition = do
     ppKeyDeposit = pp ^. ppKeyDepositL
     checkDepositAgainstPParams deposit =
       deposit == ppKeyDeposit ?! IncorrectDepositDELEG deposit
-    registerStakeCredential stakeCred dsUnified =
-      -- This looks like it should have been a right-biased union, so that the (reward, deposit) pair would be inserted
-      -- (or overwritten) in the UMap. But since we are sure that the stake credential isn't a member yet
-      -- it will still work. The reason we cannot use a right-biased union here is because UMap treats deposits specially
-      -- in right-biased unions, and is unable to accept new deposits.
```

Comment:
I don't really understand what this means:  
> UMap treats deposits specially in right-biased unions, and is unable to accept new deposits.

Do you think it's related to this function: `unionKeyDeposits`  in [Umap](https://github.com/IntersectMBO/cardano-ledger/blob/master/libs/cardano-ledger-core/src/Cardano/Ledger/UMap.hs#L927C1-L927C17) ?
It's the only place i could find that would indicate deposits being treated specially, but it doesn't look like this function is used :thinking: 

`UM.insert` makes perfect sense to me, I would love to understand the rationale from the comment, but feel free to ignore it, i was just curious.

Reply:
This comment had to do with this ugly wart that we have in the UMap that described by this comment: https://github.com/IntersectMBO/cardano-ledger/blob/8da25c0780e3b84122b2589e9c3c319046d1ab9f/libs/cardano-ledger-core/src/Cardano/Ledger/UMap.hs#L884-L887